### PR TITLE
Wrap MySql subselects for update/delete

### PR DESCRIPTION
**Pull request by garrensmith**
MySql will error if a `Update` or `Delete` query has a subselect that references a table that is being updated or deleted to get around that, we need to wrap the table in a tmp table name

```
UPDATE `crabbywilderness` SET `val` = ?
WHERE (`crabbywilderness`.`id`)
IN (SELECT `t1`.`id` FROM `crabbywilderness` AS `t1` INNER JOIN `breakabletomatoes` AS `j` ON `j`.`id` = `t1`.`id2`)
```

### CODE REVIEW

**Review by tomhoule — APPROVED** [review]
It looks like a cleaner solution than the engine-based one we discussed on zoom :+1: I think @pimeys should have a look too because I'm not as familiar with the quaint query builder.

File: src/visitor/mysql.rs
Comment (pimeys) [inline src/visitor/mysql.rs:319]: Can we have name clashes with the outer or inner queries that might make some updates/deletes broken?

**Review by pimeys — COMMENTED** [review]
Much better than the first round in the engines. There are a few places I'd consider a bit more before merging.

File: src/visitor/mysql.rs
Comment (Weakky) [inline src/visitor/mysql.rs:331]: I don't understand where that `_table` argument is used. You're either passing `None` in the main `visitor.rs` and here you're using state from the `MySql` visitor that's being set when visiting an `update` or a `delete`. Am I missing something?

File: src/visitor/mysql.rs
Comment (Weakky) [inline src/visitor/mysql.rs:15]: Would you mind either being a bit more specific on the name of the field or just adding docs that explain what this is used for? `table` is pretty generic in the context of the visitor given that it's only set when visiting an update or a delete stmt.

File: src/visitor/mysql.rs
Comment (Weakky) [reply src/visitor/mysql.rs:255]: An `Option` was likely used so that it could be initialized to something when the visitor is initialized

File: src/visitor/mysql.rs
Comment (garrensmith) [reply src/visitor/mysql.rs:331]: sorry it was suppose to be removed.

File: src/visitor/mysql.rs
Comment (garrensmith) [reply src/visitor/mysql.rs:314]: it was meant to be removed.

File: src/visitor/mysql.rs
Comment (garrensmith) [reply src/visitor/mysql.rs:319]: I don't think so. We can only update one table at a time.

File: src/visitor/mysql.rs
Comment (garrensmith) [reply src/visitor/mysql.rs:255]: Yes correct

File: src/tests/query.rs
Comment (garrensmith) [reply src/tests/query.rs]: I've changed it to check the result. All we really need to check is that it doesn't throw an error

File: src/visitor/mysql.rs
Comment (garrensmith) [reply src/visitor/mysql.rs]: removed

File: src/tests/test_api.rs
Comment (pimeys) [reply src/tests/test_api.rs:26]: Maybe name the other one as `create_temp_table` and this one `create_table`.

File: src/tests/test_api/mysql.rs
Comment (pimeys) [inline src/tests/test_api/mysql.rs:43]: Do we btw. clean these tables after a test is run, so they don't leak to the other tests?

File: src/visitor/mysql.rs
Comment (pimeys) [inline src/visitor/mysql.rs:15]:
```suggestion
    /// The table a deleting or updating query is acting on.
```

**Review by pimeys — COMMENTED** [review]
Much better. Please write one more test to `visitor/mysql.rs` unit tests to show what kind of query gets generated from a certain input. Maybe one for delete and one for update. Then this is ready to be merged!

**Review by Weakky — APPROVED** [review]
Nothing else to add on top of what Julius already suggested in his latest review! Much better indeed, great work 👍

File: src/visitor/mysql.rs
Comment (pimeys) [inline src/visitor/mysql.rs:939]: Should we also have a test for deletes? Just copy this but make it to do `DELETE` so we also test that code path.

**Review by pimeys — APPROVED** [review]
One small test more, otherwise ready to go. Approving now, you decide is the test worth it.

File: src/visitor/mysql.rs
Comment (garrensmith) [reply src/visitor/mysql.rs]: added

**Pull request merged**
